encoding/video-artifacts.md

```diff
@@ -232,6 +234,29 @@ or a manual filtering of dehalo\_alpha with dhhmask
 
 [![field-noise.jpg](images/3cnvimage103.png)](https://diff.pics/84URvW5IYSdO/1)
 
+
+### Underflow/Overflow
+
+While most of the anime produced
+use the YUV 8-bit limited range[^4],
+we occasionally find some videos having the "limited range" flag set
+while containing full range content.
+This often results in oversaturated colors and weird brightness.
+Thus, it is strongly recommended
+to check the brightness levels with `hist.Levels()`.
+
+[![Example of underflow (click for comparison)](images/underflow.jpg)](https://slowpics.org/comparison/f125e799-fdff-4c4c-8c9d-707af021bd88)
+
+[![Example of overflow (click for comparison)](images/overflow.jpg)](https://slowpics.org/comparison/6e24ffe9-e068-4f33-b2e7-639031d512f2)
+
+To fix this problem,
+simply use `mvf.Depth(input, fulls=True, fulld=False)`
+or set the "full range" flag on the video,
+so the values can be interpreted accordingly.
```

Comment:
Is there any advantage in scaling under- or overflowing video into the limited range compared to setting the full range flag? I would imagine the limited range being more widely supported by players while reducing the range obviously means reducing the available information in the video. Or what exactly does `mvf.Depth` do to begin with?

Furthermore, is this relevant when encoding with 10-bit?

Reply:
Limited range have greater compatibility while some players may ignore the "full range" flag and treat the video as a limited one.
Then we should process this problem under 16bit or even float point, there shouldn't be any noticeable loss.
Using `mvf.Depth` is a simple but not such a good way because sometimes these problems are complicated, and you won't expect any precision problems while using it.